autoload/lsp/utils.vim

```diff
@@ -245,6 +245,31 @@ function! s:get_base64_alphabet() abort
     return l:alphabet
 endfunction
 
+function! lsp#utils#get_before_line() abort
```

Comment:
Currently it is difficult to now which are public for outside vim-lsp use and which are internal due to the nature of how vim script work?

Can we prefix with `_get_before_line()` so that we only use it inside vim-lsp. We should do the same for `get_before_char_skip_white`